### PR TITLE
Implement construct_inputs for acquisition functions

### PR DESCRIPTION
**Pull request by Balandat**
Summary: Implement the `construct_inputs` method from D28021531 for most acquisition funcitons

Differential Revision: D28170202

### CODE REVIEW

File: botorch/acquisition/acquisition.py [hunk 1]
```diff
@@ -64,6 +66,37 @@ def forward(self, X: Tensor) -> Tensor:
...
+        **kwargs: Any
+    ) -> Dict[str, Any]:
+        r"""Construct kwargs for the `AcquisitionFunction` constructor.
+
+        Args:
+            model: The model to be used in the acquisition function.
+            training_data: A TrainingData object contraining the model's
+                training data. Used e.g. to extract inputs such as `best_f`
+                for expected improvement acquisition functions.
+            objective: The objective to in the acquisition function.
```
Comment (saitcakmak) [inline botorch/acquisition/acquisition.py:84]: Nice feature! Just two typos - "contraining" & "the objective to in". These repeat in all `construct_inputs` docstrings.

**Comment by saitcakmak** [comment]
I was also wondering if the `training_data` is needed as an additional argument. We could easily extract this from the model, and untransform it to get the original data where needed. I was just curious if there's a good reason for not doing it this way that I am missing

File: botorch/acquisition/analytic.py [hunk 2]
```diff
@@ -120,6 +106,53 @@ def __init__(
...
+        Returns:
+            A dict mapping kwarg names of the constructor to values.
+
+        Example:
+            >>> kwargs = ExpectedImprovement.construct_inputs(
+                    model, training_data,
+                )
+            >>> EI = ExpectedImprovement(**kwargs)
+        """
+        best_f = training_data.Y.max(-2).values.squeeze(-1)
```
Comment (saitcakmak) [inline botorch/acquisition/analytic.py:134]: If using a multi-output model with a `ScalarizedObjective`, I think we need to apply the same scalarization here before taking the max.

**Comment by Balandat** [comment]
Superseded by #788

**Pull request closed without merging**